compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1313,10 +1392,22 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
                 };
                 let mut abi = Abi::Aggregate { sized: true };
 
-                // Without latter check aligned enums with custom discriminant values
-                // Would result in ICE see the issue #92464 for more info
-                if tag.size(dl) == size || variants.iter().all(|layout| layout.is_empty()) {
+                if layout_variants.iter().all(|v| v.abi.is_uninhabited()) {
+                    abi = Abi::Uninhabited;
```

Comment:
I reordered this a bit so it better fits the pattern of these three branches here computing the `abi`.

Rewriting this without mutable variables would be cleaner but that is an undertaking for someone else. ;)